frappe/modules/patch_handler.py

```diff
@@ -204,9 +206,27 @@ def execute_patch(patchmodule: str, method=None, methodargs=None):
 	return True
 
 
-def update_patch_log(patchmodule):
+def update_patch_log(patchmodule, skipped):
 	"""update patch_file in patch log"""
-	frappe.get_doc({"doctype": "Patch Log", "patch": patchmodule}).insert(ignore_permissions=True)
+	"""
+	skipped -> Patch failed
+	not skipped -> Patch executed successfully
+	"""
+	patch_log_exists = frappe.db.get_value("Patch Log", {"patch": patchmodule, "skipped": 1})
+
+	if patch_log_exists and not skipped:
+		frappe.db.set_value("Patch Log", {"patch": patchmodule}, "skipped", 0)
+	elif patch_log_exists:
+		traceback = frappe.get_traceback(with_context=True)
+		frappe.db.set_value("Patch Log", {"patch": patchmodule}, "traceback", traceback)
+	else:
+		patch = frappe.get_doc({"doctype": "Patch Log", "patch": patchmodule})
+
+		if skipped:
+			traceback = frappe.get_traceback(with_context=True)
+			patch.skipped = 1
+			patch.traceback = traceback
+		patch.insert(ignore_permissions=True)
```

Comment:
Do we need this complex logic to update old one? 


Why not just create new log everytime. Final non-skipped one matters only because we filter skipped ones in patch handler.